src/main/scala/akka/persistence/cassandra/EventsByPersistenceIdIterator.scala

```diff
@@ -0,0 +1,84 @@
+package akka.persistence.cassandra
+
+import java.lang.{Long => JLong}
+import java.nio.ByteBuffer
+
+import scala.collection.JavaConverters._
+
+import akka.persistence.PersistentRepr
+import akka.persistence.PersistentRepr._
+import akka.persistence.cassandra.query.journal.{CassandraReadStatements, CassandraReadJournalConfig}
+import akka.serialization.Serialization
+import com.datastax.driver.core.utils.Bytes
+import com.datastax.driver.core.{ResultSet, Row, Session}
+
+final class EventsByPersistenceIdIterator(
```

Comment:
I think the iterator abstraction only adds complexity, since we must use non-blocking. We should at some point use non-blocking for ordinary replays also.